File: app/src/main/java/com/moneymanager/repo/interfaces/IAccount.java
// Created by pranay on 07/01/17.

package com.moneymanager.repo.interfaces;

import com.moneymanager.entities.Account;
import com.moneymanager.exceptions.InsufficientBalanceException;
import com.moneymanager.exceptions.NoAccountsException;

public interface IAccount {

	long insertNewAccount(Account account);

	Account[] getAllAccounts(String column, String order) throws NoAccountsException;

	Account getAccount(int id);

	void removeAccount(int id);

	double getSumOfBalanceOfAllAccounts();

	double getSumOfBalanceOfAccount(int selectedAccount);

	void updateAccountBalance(int id, double amount, boolean add) throws InsufficientBalanceException;

}

File: app/src/main/java/com/moneymanager/repo/interfaces/IAccount.kt
package com.moneymanager.repo.interfaces

import com.moneymanager.entities.Account

interface IAccount {

	fun insertNewAccount(acc: Account): Long

	fun getAllAccounts(column: String?, order: String?): Array<Account?>

	fun getAccount(id: Int): Account

	fun removeAccount(id: Int)

	fun getAllAcountsOverView(): Int

}